### ISSUE TITLE
Result with "template-like" output is broken.

### ISSUE THREAD

**Issue by akaspin** [ansible/ansible#20560]
##### ISSUE TYPE

 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.2.1.0
```

##### SUMMARY

I'm using "uri" module to register kapacitor tasks. So kapacitor tick files contains go template statements in double curly braces. Then I register result and analyse HTTP code. After upgrade result analysing is broken:

```
fatal: [influxdb-01]: FAILED! => {
    "failed": true, 
    "msg": "{'msg': u'One or more items failed', 'failed': True, 'changed': False, 'results': [{u'status': -1, '_ansible_parsed': True, '_ansible_no_log': False, u'failed': True, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', '_ansible_item_result': True, u'content': u'', 'item': u'/Users/spin/Craft/applift/borderline/ansible-old/roles/kapacitor-tasks/files/cpu_alert', 'invocation': {'module_name': u'uri', u'module_args': {u'directory_mode': None, u'force': False, u'remote_src': None, u'status_code': [u'200'], u'follow': False, u'owner': None, u'body_format': u'raw', u'group': None, u'use_proxy': True, u'unsafe_writes': None, u'setype': None, u'content': None, u'serole': None, u'follow_redirects': u'all', u'return_content': False, u'method': u'POST', u'body': {u'status': u'enabled', u'type': u'stream', u'id': u'cpu_alert', u'dbrps': [{u'rp': u'autogen', u'db': u'metrics-stage-01'}], u'script': u'var info = 70\\nvar warn = 80\\nvar crit = 90\\nvar period = 10s\\nvar every = 10s\\n\\nvar data = stream\\n  |from()\\n    .measurement(\\'cpu\\')\\n    .groupBy(\\'host\\')\\n    .where(lambda: \"cpu\" == \\'cpu-total\\')\\n  |window()\\n    .period(period)\\n    .every(every)\\n  |mean(\\'usage_user\\')\\n    .as(\\'stat\\')\\nvar alert = data\\n  |alert()\\n    .id(\\'{{ index .Tags \"host\"}}/cpu_used\\')\\n    .message(\\'{{ .ID }}: {{ index .Fields \"stat\" }}\\')\\n    .info(lambda: \"stat\" > info)\\n    .warn(lambda: \"stat\" > warn)\\n    .crit(lambda: \"stat\" > crit)\\nalert\\n  .slack()\\n    .username(\\'cpu\\')'}, u'timeout': 30, u'src': None, u'dest': None, u'selevel': None, u'force_basic_auth': False, u'removes': None, u'http_agent': u'ansible-httpget', u'regexp': None, u'url_password': None, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', u'validate_certs': True, u'seuser': None, u'creates': None, u'headers': {}, u'delimiter': None, u'mode': None, u'url_username': None, u'backup': None}}, u'redirected': False, u'msg': u'Status code was not [200]: An unknown error occurred: must be string or buffer, not dict'}, {u'status': -1, '_ansible_parsed': True, '_ansible_no_log': False, u'failed': True, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', '_ansible_item_result': True, u'content': u'', 'item': u'/Users/spin/Craft/applift/borderline/ansible-old/roles/kapacitor-tasks/files/disk_alert', 'invocation': {'module_name': u'uri', u'module_args': {u'directory_mode': None, u'force': False, u'remote_src': None, u'status_code': [u'200'], u'follow': False, u'owner': None, u'body_format': u'raw', u'group': None, u'use_proxy': True, u'unsafe_writes': None, u'setype': None, u'content': None, u'serole': None, u'follow_redirects': u'all', u'return_content': False, u'method': u'POST', u'body': {u'status': u'enabled', u'type': u'stream', u'id': u'disk_alert', u'dbrps': [{u'rp': u'autogen', u'db': u'metrics-stage-01'}], u'script': u'var info = 75\\nvar warn = 85\\nvar crit = 92\\nvar period = 10s\\nvar every = 10s\\n\\nvar data = stream\\n    |from()\\n        .database(\\'telegraf\\')\\n        .retentionPolicy(\\'autogen\\')\\n        .measurement(\\'disk\\')\\n        .groupBy(\\'host\\')\\n    |window()\\n        .period(period)\\n        .every(every)\\n    |mean(\\'used_percent\\')\\n        .as(\\'stat\\')\\nvar alert = data\\n    |alert()\\n        .id(\\'{{ index .Tags \"host\"}}/disk_used\\')\\n        .message(\\'{{ .ID }}:{{ index .Fields \"stat\" }}\\')\\n        .info(lambda: \"stat\" > info)\\n        .warn(lambda: \"stat\" > warn)\\n        .crit(lambda: \"stat\" > crit)\\n\\nalert\\n  .slack()\\n    .username(\\'disk\\')'}, u'timeout': 30, u'src': None, u'dest': None, u'selevel': None, u'force_basic_auth': False, u'removes': None, u'http_agent': u'ansible-httpget', u'regexp': None, u'url_password': None, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', u'validate_certs': True, u'seuser': None, u'creates': None, u'headers': {}, u'delimiter': None, u'mode': None, u'url_username': None, u'backup': None}}, u'redirected': False, u'msg': u'Status code was not [200]: An unknown error occurred: must be string or buffer, not dict'}, {u'status': -1, '_ansible_parsed': True, '_ansible_no_log': False, u'failed': True, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', '_ansible_item_result': True, u'content': u'', 'item': u'/Users/spin/Craft/applift/borderline/ansible-old/roles/kapacitor-tasks/files/mem_alert', 'invocation': {'module_name': u'uri', u'module_args': {u'directory_mode': None, u'force': False, u'remote_src': None, u'status_code': [u'200'], u'follow': False, u'owner': None, u'body_format': u'raw', u'group': None, u'use_proxy': True, u'unsafe_writes': None, u'setype': None, u'content': None, u'serole': None, u'follow_redirects': u'all', u'return_content': False, u'method': u'POST', u'body': {u'status': u'enabled', u'type': u'stream', u'id': u'mem_alert', u'dbrps': [{u'rp': u'autogen', u'db': u'metrics-stage-01'}], u'script': u'var info = 70\\nvar warn = 85\\nvar crit = 92\\nvar period = 10s\\nvar every = 10s\\n\\nvar data = stream\\n  |from()\\n    .database(\\'telegraf\\')\\n    .retentionPolicy(\\'autogen\\')\\n    .measurement(\\'mem\\')\\n    .groupBy(\\'host\\')\\n  |window()\\n    .period(period)\\n    .every(every)\\n  |mean(\\'used_percent\\')\\n    .as(\\'stat\\')\\n    \\nvar alert = data\\n  |alert()\\n    .id(\\'{{ index .Tags \"host\"}}/mem_used\\')\\n    .message(\\'{{ .ID }}:{{ index .Fields \"stat\" }}\\')\\n    .info(lambda: \"stat\" > info)\\n    .warn(lambda: \"stat\" > warn)\\n    .crit(lambda: \"stat\" > crit)\\nalert\\n  .slack()\\n    .username(\\'mem\\')'}, u'timeout': 30, u'src': None, u'dest': None, u'selevel': None, u'force_basic_auth': False, u'removes': None, u'http_agent': u'ansible-httpget', u'regexp': None, u'url_password': None, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', u'validate_certs': True, u'seuser': None, u'creates': None, u'headers': {}, u'delimiter': None, u'mode': None, u'url_username': None, u'backup': None}}, u'redirected': False, u'msg': u'Status code was not [200]: An unknown error occurred: must be string or buffer, not dict'}]}: template error while templating string: expected token 'end of print statement', got 'string'. String: var info = 70\nvar warn = 80\nvar crit = 90\nvar period = 10s\nvar every = 10s\n\nvar data = stream\n  |from()\n    .measurement('cpu')\n    .groupBy('host')\n    .where(lambda: \"cpu\" == 'cpu-total')\n  |window()\n    .period(period)\n    .every(every)\n  |mean('usage_user')\n    .as('stat')\nvar alert = data\n  |alert()\n    .id('{{ index .Tags \"host\"}}/cpu_used')\n    .message('{{ .ID }}: {{ index .Fields \"stat\" }}')\n    .info(lambda: \"stat\" > info)\n    .warn(lambda: \"stat\" > warn)\n    .crit(lambda: \"stat\" > crit)\nalert\n  .slack()\n    .username('cpu')"
}
```

**Comment by alikins** [comment]
For easier reading, I'm including a version of the output from the report that has had \n's unescaped and a little untested formatting here.


```
fatal: [influxdb-01]: FAILED! => {
    "failed": true,
    "msg": "{'msg': u'One or more items failed',
             'failed': True,
             'changed': False,
             'results': [{u'status': -1,
                          '_ansible_parsed': True,
                          '_ansible_no_log': False,
                          u'failed': True,
                          u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks',
                          '_ansible_item_result': True,
                          u'content': u'',
                          'item': u'/Users/spin/Craft/applift/borderline/ansible-old/roles/kapacitor-tasks/files/cpu_alert',
                          'invocation': {'module_name': u'uri',
                                          u'module_args': {u'directory_mode': None,
                                                           u'force': False,
                                                           u'remote_src': None,
                                                           u'status_code': [u'200'],
                                                           u'follow': False,
                                                           u'owner': None,
                                                           u'body_format': u'raw',
                                                           u'group': None,
                                                           u'use_proxy': True,
                                                           u'unsafe_writes': None,
                                                           u'setype': None,
                                                           u'content': None,
                                                           u'serole': None,
                                                           u'follow_redirects': u'all',
                                                           u'return_content': False,
                                                           u'method': u'POST',
                                                           u'body': {u'status': u'enabled',
                                                                     u'type': u'stream',
                                                                     u'id': u'cpu_alert',
                                                                     u'dbrps': [{u'rp': u'autogen', u'db': u'metrics-stage-01'}],
                                                                     u'script':
u'var info = 70
var warn = 80
var crit = 90
var period = 10s
var every = 10s

var data = stream
  |from()
    .measurement('cpu')
    .groupBy('host')
    .where(lambda: "cpu" == 'cpu-total')
  |window()
    .period(period)
    .every(every)
  |mean('usage_user')
    .as('stat')
var alert = data
  |alert()
    .id('{{ index .Tags "host"}}/cpu_used')
    .message('{{ .ID }}: {{ index .Fields "stat" }}')
    .info(lambda: "stat" > info)
    .warn(lambda: "stat" > warn)
    .crit(lambda: "stat" > crit)
alert
  .slack()
    .username('cpu')'}, u'timeout': 30, u'src': None, u'dest': None, u'selevel': None, u'force_basic_auth': False, u'removes': None, u'http_agent': u'ansible-httpget', u'regexp': None, u'url_password': None, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', u'validate_certs': True, u'seuser': None, u'creates': None, u'headers': {}, u'delimiter': None, u'mode': None, u'url_username': None, u'backup': None}}, u'redirected': False, u'msg': u'Status code was not [200]: An unknown error occurred: must be string or buffer, not dict'}, {u'status': -1, '_ansible_parsed': True, '_ansible_no_log': False, u'failed': True, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', '_ansible_item_result': True, u'content': u'', 'item': u'/Users/spin/Craft/applift/borderline/ansible-old/roles/kapacitor-tasks/files/disk_alert', 'invocation': {'module_name': u'uri', u'module_args': {u'directory_mode': None, u'force': False, u'remote_src': None, u'status_code': [u'200'], u'follow': False, u'owner': None, u'body_format': u'raw', u'group': None, u'use_proxy': True, u'unsafe_writes': None, u'setype': None, u'content': None, u'serole': None, u'follow_redirects': u'all', u'return_content': False, u'method': u'POST', u'body': {u'status': u'enabled', u'type': u'stream', u'id': u'disk_alert', u'dbrps': [{u'rp': u'autogen', u'db': u'metrics-stage-01'}], u'script': u'var info = 75
var warn = 85
var crit = 92
var period = 10s
var every = 10s

var data = stream
    |from()
        .database('telegraf')
        .retentionPolicy('autogen')
        .measurement('disk')
        .groupBy('host')
    |window()
        .period(period)
        .every(every)
    |mean('used_percent')
        .as('stat')
var alert = data
    |alert()
        .id('{{ index .Tags "host"}}/disk_used')
        .message('{{ .ID }}:{{ index .Fields "stat" }}')
        .info(lambda: "stat" > info)
        .warn(lambda: "stat" > warn)
        .crit(lambda: "stat" > crit)

alert
  .slack()
    .username('disk')'}, u'timeout': 30, u'src': None, u'dest': None, u'selevel': None, u'force_basic_auth': False, u'removes': None, u'http_agent': u'ansible-httpget', u'regexp': None, u'url_password': None, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', u'validate_certs': True, u'seuser': None, u'creates': None, u'headers': {}, u'delimiter': None, u'mode': None, u'url_username': None, u'backup': None}}, u'redirected': False, u'msg': u'Status code was not [200]: An unknown error occurred: must be string or buffer, not dict'}, {u'status': -1, '_ansible_parsed': True, '_ansible_no_log': False, u'failed': True, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', '_ansible_item_result': True, u'content': u'', 'item': u'/Users/spin/Craft/applift/borderline/ansible-old/roles/kapacitor-tasks/files/mem_alert', 'invocation': {'module_name': u'uri', u'module_args': {u'directory_mode': None, u'force': False, u'remote_src': None, u'status_code': [u'200'], u'follow': False, u'owner': None, u'body_format': u'raw', u'group': None, u'use_proxy': True, u'unsafe_writes': None, u'setype': None, u'content': None, u'serole': None, u'follow_redirects': u'all', u'return_content': False, u'method': u'POST', u'body': {u'status': u'enabled', u'type': u'stream', u'id': u'mem_alert', u'dbrps': [{u'rp': u'autogen', u'db': u'metrics-stage-01'}], u'script': u'var info = 70
var warn = 85
var crit = 92
var period = 10s
var every = 10s

var data = stream
  |from()
    .database('telegraf')
    .retentionPolicy('autogen')
    .measurement('mem')
    .groupBy('host')
  |window()
    .period(period)
    .every(every)
  |mean('used_percent')
    .as('stat')

var alert = data
  |alert()
    .id('{{ index .Tags "host"}}/mem_used')
    .message('{{ .ID }}:{{ index .Fields "stat" }}')
    .info(lambda: "stat" > info)
    .warn(lambda: "stat" > warn)
    .crit(lambda: "stat" > crit)
alert
  .slack()
    .username('mem')'}, u'timeout': 30, u'src': None, u'dest': None, u'selevel': None, u'force_basic_auth': False, u'removes': None, u'http_agent': u'ansible-httpget', u'regexp': None, u'url_password': None, u'url': u'http://kapacitor.service.stage-01:9092/kapacitor/v1/tasks', u'validate_certs': True, u'seuser': None, u'creates': None, u'headers': {}, u'delimiter': None, u'mode': None, u'url_username': None, u'backup': None}}, u'redirected': False, u'msg': u'Status code was not [200]: An unknown error occurred: must be string or buffer, not dict'}]}: template error while templating string: expected token 'end of print statement', got 'string'. String: var info = 70
var warn = 80
var crit = 90
var period = 10s
var every = 10s
var data = stream
  |from()
    .measurement('cpu')
    .groupBy('host')
    .where(lambda: "cpu" == 'cpu-total')
  |window()
    .period(period)
    .every(every)
  |mean('usage_user')
    .as('stat')

var alert = data
  |alert()
    .id('{{ index .Tags "host"}}/cpu_used')
    .message('{{ .ID }}: {{ index .Fields "stat" }}')
    .info(lambda: "stat" > info)
    .warn(lambda: "stat" > warn)
    .crit(lambda: "stat" > crit)
alert
  .slack()
    .username('cpu')"
}

```

**Comment by alikins** [comment]
> Then I register result and analyse HTTP code.

Whats the playbook data that invokes this task look like?

**Comment by alikins** [comment]
Was this working with ansible 2.2.0 (or other versions?)

**Comment by jimi-c** [comment]
This should be a dupe of #20568, which was fixed. If not, please let us know.

Thanks!

**Comment by akaspin** [comment]
@jimi-c Still happens in 2.2.1.0

**Comment by evgeni** [comment]
Is this maybe the same as [CVE-2017-7466](https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2017-7466)?

Fixed in 2.2: https://github.com/ansible/ansible/commit/7ff9fa52cfcef2065f0db80d85dd94b9b754839c
Fixed in 2.3/devel: https://github.com/ansible/ansible/commit/0d418789a298561fded9bce977d34babc9097079

**Comment by sivel** [comment]
I can not reproduce this issue in current versions of Ansible. If you are still having problems, please feel free to open a new issue with a reproducer.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project
